docs/src/_includes/components/rule.macro.html

```diff
@@ -18,7 +18,7 @@
             <span class="rule__status">removed</span>
         </p>
         {%- if params.replacedBy -%}
-        <p class="rule__description">Replaced by {{ ruleList({ rules: params.replacedBy }) }}</p>
+        <p class="rule__description">Replaced by {{ replacementRuleList({ specifiers: params.replacedBy }) }}</p>
```

Comment:
Replacement rules for removed ones are not rendered:

![image](https://github.com/user-attachments/assets/52c6247f-3e4d-486b-9a2b-42c10fef96e8)

https://deploy-preview-19238--docs-eslint.netlify.app/rules/#removed

I think it's because `replacedBy` is in the old format (array of strings). We could update the data in [`rule-type-list.json`](https://github.com/eslint/eslint/blob/main/conf/rule-type-list.json) to use the new format, e.g., `["generator-star-spacing"]` -> `[{ "rule": { "name": "generator-star-spacing" } }]`